undefined reference to `af_save_image_memory' when freeimage is absent

So when compiling the newest release (also happens on git master) I get this:

    [ 45%] Linking CXX executable approx1_cpu
    ../src/backend/cpu/libafcpu.so.3.2.0: undefined reference to `af_save_image_memory'

Using gcc 5.2.

make verbose=1 output is:

    [ 45%] Linking CXX executable approx1_cpu
    cd /home/svenstaro/src/arrayfire/build/test && /usr/bin/cmake -E cmake_link_script    CMakeFiles/approx1_cpu.dir/link.txt --verbose=1
    /usr/bin/c++   -O3 -DNDEBUG   CMakeFiles/approx1_cpu.dir/approx1.cpp.o  -o approx1_cpu -rdynamic ../src/backend/cpu/libafcpu.so.3.2.0 -pthread ../third_party/gtest/src/googletest-build/./libgtest.a   ../third_party/gtest/src/googletest-build/./libgtest_main.a -Wl,- rpath,/home/svenstaro/src/arrayfire/build/src/backend/cpu:/home/svenstaro/src/arrayfire/build/third_party/ forge/src/forge-ext-build/src -Wl,-rpath-link,/home/svenstaro/src/arrayfire/build/third_party/forge/src/forge- ext-build/src

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -148,11 +148,6 @@ ENDIF()
 # Architechture Definitions
 INCLUDE(${CMAKE_MODULE_PATH}/TargetArch.cmake)
 target_architecture(ARCH)
-IF(${ARCH} STREQUAL "x86_64")
-    ADD_DEFINITIONS(-DARCH_64)
-ELSE(${ARCH})
-    ADD_DEFINITIONS(-DARCH_32)
-ENDIF()
 
 INCLUDE(${CMAKE_MODULE_PATH}/Version.cmake)
 

--- a/src/api/c/imageio.cpp
+++ b/src/api/c/imageio.cpp
@@ -760,7 +760,7 @@ af_err af_load_image_memory(af_array *out, const void* ptr)
     return AF_ERR_NOT_CONFIGURED;
 }
 
-af_err af_save_image_memory(void **ptr, const char* filename, const af_array in_)
+af_err af_save_image_memory(void **ptr, const af_array in_, const af_image_format format)
 {
     printf("Error: Image IO requires FreeImage. See https://github.com/arrayfire/arrayfire\n");
     return AF_ERR_NOT_CONFIGURED;

--- a/src/backend/cpu/math.cpp
+++ b/src/backend/cpu/math.cpp
@@ -14,9 +14,6 @@ namespace cpu
     uint abs(uint val) { return val; }
     uchar abs(uchar val) { return val; }
     uintl abs(uintl val) { return val; }
-#if !(defined(OS_WIN) || (defined(ARCH_32) && defined(OS_LNX)))  // Not(Windows or Tegra)
-    size_t abs(size_t val) { return val; }
-#endif
 
     cfloat  scalar(float val)
     {

--- a/src/backend/cpu/math.hpp
+++ b/src/backend/cpu/math.hpp
@@ -20,9 +20,6 @@ namespace cpu
     uint abs(uint val);
     uchar abs(uchar val);
     uintl abs(uintl val);
-#if !(defined(OS_WIN) || (defined(ARCH_32) && defined(OS_LNX)))  // Not(Windows or Tegra)
-    size_t abs(size_t val);
-#endif
 
     template<typename T> static inline T min(T lhs, T rhs) { return std::min(lhs, rhs); }
     cfloat min(cfloat lhs, cfloat rhs);

--- a/src/backend/cpu/platform.cpp
+++ b/src/backend/cpu/platform.cpp
@@ -172,15 +172,11 @@ CPUInfo::CPUInfo()
 namespace cpu
 {
 
-static const char *get_system(void)
+static const std::string get_system(void)
 {
-    return
-#if defined(ARCH_32)
-    "32-bit "
-#elif defined(ARCH_64)
-    "64-bit "
-#endif
+    std::string arch = (sizeof(void *) == 4) ? "32-bit " : "64-bit ";
 
+    return arch +
 #if defined(OS_LNX)
     "Linux";
 #elif defined(OS_WIN)

--- a/src/backend/cuda/platform.cpp
+++ b/src/backend/cuda/platform.cpp
@@ -116,15 +116,11 @@ static inline bool card_compare_num(const cudaDevice_t &l, const cudaDevice_t &r
     return 0;
 }
 
-static const char *get_system(void)
+static const std::string get_system(void)
 {
-    return
-#if defined(ARCH_32)
-    "32-bit "
-#elif defined(ARCH_64)
-    "64-bit "
-#endif
+    std::string arch = (sizeof(void *) == 4) ? "32-bit " : "64-bit ";
 
+    return arch +
 #if defined(OS_LNX)
     "Linux";
 #elif defined(OS_WIN)

--- a/src/backend/opencl/platform.cpp
+++ b/src/backend/opencl/platform.cpp
@@ -60,15 +60,11 @@ static const std::string CL_GL_SHARING_EXT = "cl_APPLE_gl_sharing";
 static const std::string CL_GL_SHARING_EXT = "cl_khr_gl_sharing";
 #endif
 
-static const char *get_system(void)
+static const std::string get_system(void)
 {
-    return
-#if defined(ARCH_32)
-    "32-bit "
-#elif defined(ARCH_64)
-    "64-bit "
-#endif
+    std::string arch = (sizeof(void *) == 4) ? "32-bit " : "64-bit ";
 
+    return arch +
 #if defined(OS_LNX)
     "Linux";
 #elif defined(OS_WIN)

